app/main/views.py

```diff
@@ -15,3 +20,46 @@ def about():
     editable_html_obj = EditableHTML.get_editable_html('about')
     return render_template(
         'main/about.html', editable_html_obj=editable_html_obj)
+
+# generates all signed URL for AWS upload
+@main.route('/sign-s3/')
+def sign_s3():
+    # Load necessary information into the application
+    S3_BUCKET = os.environ.get('S3_BUCKET')
+    S3_REGION = os.environ.get('S3_REGION')
+    TARGET_FOLDER = 'json/'
+    # Load required data from the request
+    pre_file_name = request.args.get('file-name')
```

Comment:
We should do a check for `pre_file_name is None` – if I try to access `/sign-s3/`, we should 404 instead of having the app crash.